### DISCUSSION TITLE
Dockerize Anvil

### DISCUSSION THREAD

**Discussion by h4shr4t3** [foundry-rs/foundry#7151] [Help]
Hello,

I'd like to run an instance of anvil inside a docker container.

As I want to start the anvil docker container along with some other containers via a docker-compose.yml file, I need a simple Dockerfile that spins up anvil as standard ENTRYPOINT.

Sth. like:
```

# Use the latest foundry image
FROM ghcr.io/foundry-rs/foundry

# Set the working directory inside the container
WORKDIR /Anvil

# Copy our source code into the container
COPY . .

# Use array syntax for ENTRYPOINT
ENTRYPOINT ["anvil", "--fork-url", "[RPC_URL]", "--fork-block-number", "19233900", "--port", "7171", "--host", "127.0.0.1", "--chain-id", "1337", "--steps-tracing", "--block-time", "12"]
```

Unfortunately this does not work as expected. I get a warning about the image's platform:

WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested

And an error:

error: unexpected argument '--fork-url [RPC-URL] --fork-block-number 19233900 --port 7171 --host 127.0.0.1 --chain-id 1337 --steps-tracing --block-time 12' found

Usage: anvil [OPTIONS] [COMMAND]

Is there a better way to set this up?

**Comment by antonis19** [comment]
I think there might be an issue with your ENTRYPOINT command and/or environment variables.

Can you try the following Dockerfile:

```Dockerfile
# syntax=docker/dockerfile:1.4
FROM ghcr.io/foundry-rs/foundry

WORKDIR /anvil

ENTRYPOINT anvil --fork-url $FORK_URL --host 0.0.0.0
```

where `FORK_URL` is an environment variable?

**Reply by h4shr4t3** [reply]
This works for me. Thanks! 
One mistake I made was that I just tried to start the container with "docker run anvil" - without providing the [OPTIONS] and [COMMAND] for starting anvil. So anvil was just missing the arguments. I thought that if I "define" the complete command within the [ENTRYPOINT] that it will start without providing the arguments from the outside. But maybe I was just not using the right syntax for it.
